src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java

```diff
@@ -368,35 +363,100 @@ public TeamMember addProjectTeamMember(final Integer projectId, final String use
    * @param project the project
    * @return ProjectPackageSummary
    */
-  private Optional<ProjectPackageSummary> convertProjectToProjectPackageSummary(
-      final Project project, final List<ProcurementProject> projects) {
+  private Optional<ProjectPackageSummary> convertProjectToProjectPackageSummary(final ProjectUserMapping project) {
     // Get Project from database
-    var dbProject = getProject(projects, project.getTender().getTenderCode());
     var projectPackageSummary = new ProjectPackageSummary();
-    if (dbProject.isPresent()) {
-      var dbEvent = getCurrentEvent(dbProject.get());
-
+      var agreementNo = project.getProject().getCaNumber();
+      var dbEvent = getCurrentEvent(project.getProject());
+      // TODO make single call instead of 2
+      try {
+        var agreementDetails = agreementsService.getAgreementDetails(agreementNo);
+        var lotDetails = agreementsService.getLotDetails(agreementNo, project.getProject().getLotNumber());
+        projectPackageSummary.setAgreementName(agreementDetails.getName());
+        projectPackageSummary.setLotName(lotDetails.getName());
+      }catch (Exception e) {
```

Comment:
Do we need this try-catch (it doesn't seem to be adding anything)?

Reply:
this is to avoid resource not found exception from agreement service.